File: engine/src/main/java/de/hanno/hpengine/engine/graphics/shader/GeometryShader.java
package de.hanno.hpengine.engine.graphics.shader;

import de.hanno.hpengine.util.ressources.CodeSource;

public class GeometryShader extends AbstractShader {

    public static GeometryShader load(OpenGlProgramManager programManager, CodeSource sourceCode) {
        return programManager.loadShader(GeometryShader.class, sourceCode);
    }

    @Override
    public OpenGLShader getShaderType() {
        return OpenGLShader.GeometryShader;
    }
}

File: engine/src/main/java/de/hanno/hpengine/engine/graphics/shader/GeometryShader.kt
package de.hanno.hpengine.engine.graphics.shader

import de.hanno.hpengine.engine.graphics.shader.Shader.ShaderType
import de.hanno.hpengine.engine.graphics.shader.define.Defines
import de.hanno.hpengine.util.ressources.CodeSource
import de.hanno.hpengine.util.ressources.FileBasedCodeSource

class GeometryShader private constructor(override var shaderSource: CodeSource, override var id: Int) : Shader {
    override val shaderType: ShaderType = ShaderType.GeometryShader

    companion object {
        fun load(programManager: OpenGlProgramManager, sourceCode: FileBasedCodeSource, defines: Defines = Defines()): GeometryShader {
            return GeometryShader(sourceCode, programManager.loadShader(ShaderType.GeometryShader, sourceCode, defines))
        }
    }
}